lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -451,8 +451,20 @@ resolveDeclRefExpr(UnresolvedDeclRefExpr *UDRE, DeclContext *DC) {
 
     performTypoCorrection(DC, UDRE->getRefKind(), Type(), Name, Loc,
                           lookupOptions, Lookup);
+    
+    // Check if any lookups share a basename. If that's the case, we need
+    // to disambiguate the diagnostic by printing the UDRE's full name.
```

Comment:
Okay, so I'll change the behavior system-wide for DeclName and then elect to use the base name in this case? I'm gonna do a pass over the other diagnostics that deal with DeclNames too